src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -285,31 +285,46 @@ struct Machine {
     nvdimm: bool,
 
     is_nvdimm_supported: bool,
+    memory_backend: Option<String>,
 }
 
 impl Machine {
-    fn new(config: &HypervisorConfig) -> Machine {
-        #[cfg(any(
-            target_arch = "aarch64",
-            target_arch = "powerpc64",
-            target_arch = "x86",
-            target_arch = "x86_64",
-        ))]
-        let is_nvdimm_supported = config.machine_info.machine_type != "microvm";
-        #[cfg(not(any(
-            target_arch = "aarch64",
-            target_arch = "powerpc64",
-            target_arch = "x86",
-            target_arch = "x86_64",
-        )))]
-        let is_nvdimm_supported = false;
+    fn new(config: Option<&HypervisorConfig>, memory_backend: Option<&str>) -> Machine {
+        let (is_nvdimm_supported, r#type, accel, options) = if let Some(config) = config {
+            #[cfg(any(
+                target_arch = "aarch64",
+                target_arch = "powerpc64",
+                target_arch = "x86",
+                target_arch = "x86_64",
+            ))]
+            let is_nvdimm_supported = config.machine_info.machine_type != "microvm";
+            #[cfg(not(any(
+                target_arch = "aarch64",
+                target_arch = "powerpc64",
+                target_arch = "x86",
+                target_arch = "x86_64",
+            )))]
+            let is_nvdimm_supported = false;
+
+            (
+                is_nvdimm_supported,
+                config.machine_info.machine_type.clone(),
+                "kvm".to_owned(),
+                config.machine_info.machine_accelerators.clone(),
+            )
+        } else {
+            (false, "".to_owned(), "".to_owned(), "".to_owned())
+        };
+
+        let memory_backend = memory_backend.map(|memory_backend| memory_backend.to_owned());
 
         Machine {
-            r#type: config.machine_info.machine_type.clone(),
-            accel: "kvm".to_owned(),
-            options: config.machine_info.machine_accelerators.clone(),
+            r#type,
+            accel,
+            options,
             nvdimm: false,
             is_nvdimm_supported,
+            memory_backend,
```

Comment:
This whole hunk seems to imply that `HypervisiorConfig` (that is, essentially, `configuration.toml`) has to be avoided on s390x at all cost.  Why is that, why `configuration.toml`'s machine type or accelerators must not be used on s390x?

Reply:
Updated based on your advice below:

---

1. Add `memory_backend: Option<String>` to `Machine`
2. Add `Machine::set_memory_backend()` along the lines of

```
fn set_memory_backend(&mut self, backend_file_id: &str) -> &mut Self {
        if !backend_file_id.is_empty() {
            self.memory_backend = Some(backend_file_id.to_owned());
        }
        self
    }
```

3. Add something like:

```
if let Some(memory_backend) = &self.memory_backend {
    params.push(format!("memory-backend={}", memory_backend.clone()));
}
```

to Machine::qemu_params()
4. Lastly, in `add_virtiofs_share()` the CCW arm should be just

```
    VirtioBusType::Ccw => self.machine.set_memory_backend(&mem_file.id)
```

Reply:
Thanks @BbolroC , you're very kind!